pyuvdata/uvdata/initializers.py

```diff
@@ -0,0 +1,579 @@
+                raise ValueError(
+                    f"nsample array shape {nsample_array.shape} does not match expected"
+                    f" shape {shape}."
+                )
+            obj.nsample_array = nsample_array
+        else:
+            obj.nsample_array = np.ones(shape, dtype=float)
+
+    return obj
+
+
+def get_antenna_params(
+    antenna_positions: np.ndarray | dict[str | int, np.ndarray],
+    antenna_names: list[str] | None = None,
+    antenna_numbers: list[int] | None = None,
+    antname_format: str = "{0:03d}",
+) -> tuple[np.ndarray, list[str], list[int]]:
+    """Configure antenna parameters for new UVData object."""
+    # Get Antenna Parameters
+    if isinstance(antenna_positions, dict):
+        if isinstance(next(iter(antenna_positions.keys())), int):
+            antenna_numbers = list(antenna_positions.keys())
+        else:
+            antenna_names = list(antenna_positions.keys())
+        antenna_positions = np.array(list(antenna_positions.values()))
+
+    if antenna_numbers is None and antenna_names is None:
+        raise ValueError(
+            "Either antenna_numbers or antenna_names must be provided unless "
+            "antenna_positions is a dict."
+        )
+
+    if antenna_names is None:
+        antenna_names = [antname_format.format(i) for i in antenna_numbers]
+    elif antenna_numbers is None:
+        try:
+            antenna_numbers = [int(name) for name in antenna_names]
+        except ValueError as e:
+            raise ValueError(
+                "Antenna names must be integers if antenna_numbers is not provided."
+            ) from e
+
+    if not isinstance(antenna_positions, np.ndarray):
+        raise ValueError("antenna_positions must be a numpy array or a dictionary.")
+
+    if antenna_positions.shape != (len(antenna_numbers), 3):
+        raise ValueError(
+            "antenna_positions must be a 2D array with shape (N_antennas, 3), "
+            f"got {antenna_positions.shape}"
+        )
+
+    if len(antenna_names) != len(set(antenna_names)):
+        raise ValueError("Duplicate antenna names found.")
+
+    if len(antenna_numbers) != len(set(antenna_numbers)):
+        raise ValueError("Duplicate antenna numbers found.")
+
+    if len(antenna_numbers) != len(antenna_names):
+        raise ValueError("antenna_numbers and antenna_names must have the same length.")
+
+    return antenna_positions, antenna_names, antenna_numbers
+
+
+def get_time_params(
+    telescope_location: Locations,
+    time_array: np.ndarray,
+    integration_time: float | np.ndarray | None = None,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Configure time parameters for new UVData object."""
+    if not isinstance(time_array, np.ndarray):
+        raise ValueError("time_array must be a numpy array.")
+
+    t = Time(time_array, format="jd", scale="utc", location=telescope_location)
+    lst_array = t.sidereal_time("apparent").radian
```

Comment:
There's a bit of a catch here because the user is supposed to input either an `EarthLocation` or `MoonLocation` as the `telescope_location` (instead of lat/lon/alt), so I have to then *unpack* that location to use `get_lst_for_time`. But as I don't expect the user to pass a `Time` object for the times, this means only really earth locations are supported. So, I'll do the unpacking here.